server/src/main/java/com/cloud/api/query/dao/SnapshotJoinDaoImpl.java

```diff
@@ -124,6 +129,16 @@ public SnapshotResponse newSnapshotResponse(ResponseObject.ResponseView view, bo
             snapshotResponse.setVolumeName(snapshot.getVolumeName());
             snapshotResponse.setVolumeType(snapshot.getVolumeType().name());
             snapshotResponse.setVirtualSize(snapshot.getVolumeSize());
+            VolumeVO volume = ApiDBUtils.findVolumeById(snapshot.getVolumeId());
+            if (volume != null && volume.getVolumeType() == Type.ROOT && volume.getInstanceId() != null) {
+                VMInstanceVO vm = ApiDBUtils.findVMInstanceById(volume.getInstanceId());
+                if (vm != null) {
+                    GuestOS guestOS = ApiDBUtils.findGuestOSById(vm.getGuestOSId());
+                    if (guestOS != null) {
+                        snapshotResponse.setOsTypeId(guestOS.getUuid());
+                    }
+                }
+            }
```

Comment:
@slavkap , you are adding a value from the volume (which is already in the join dao) , does it make sense to let the view do this work? (i.e. add guestos to the join)

Reply:
Hi @slavkap can you check this ^^^

Reply:
hey @DaanHoogland, sorry for the late response! I've checked the volume view and it doesn't have the required fields. Also, if I add the quest OS there, this will require more changes and updates of the DB for the already created volumes